For issue #56

GrubbyZebra opened the pull request:
Here are the files I've changed/added for the new Kerbin Programs.

Made 2 small changes to the strategia source code to support whatever homeworld the player has set up. 

Included the strategy and contract files. There is a strategy for both unmanned and manned exploration. The unmanned strategy is only available before orbiting Kerbin.

GrubbyZebra committed:
Add files via upload

GrubbyZebra committed:
Delete UncrewedMissions-1-KerbinProbes.cfg

GrubbyZebra committed:
Delete KerbinProbes.cfg

GrubbyZebra committed:
Rename CrewedMissions-1-MoonProgram.cfg to CrewedMissions-2-MoonProgram.cfg

GrubbyZebra committed:
Rename CrewedMissions-2-PlanetaryProgram.cfg to CrewedMissions-3-PlanetaryProgram.cfg

GrubbyZebra committed:
Rename CrewedMissions-3-GasGiantProgram.cfg to CrewedMissions-4-GasGiantProgram.cfg

GrubbyZebra committed:
Rename UncrewedMissions-4-FlyBys.cfg to UncrewedMissions-5-FlyBys.cfg

GrubbyZebra committed:
Add files via upload

GrubbyZebra committed:
Delete KerbinProbes.cfg

GrubbyZebra committed:
Rename UncrewedMissions-3-ImpactorProbes.cfg to UncrewedMissions-4-ImpactorProbes.cfg

GrubbyZebra committed:
Rename UncrewedMissions-2-PlanetaryProbes.cfg to UncrewedMissions-3-PlanetaryProbes.cfg

GrubbyZebra committed:
Rename UncrewedMissions-1-MoonProbes.cfg to UncrewedMissions-2-MoonProbes.cfg

GrubbyZebra committed:
Add files via upload

GrubbyZebra committed:
Add files via upload

GrubbyZebra committed:
Update KerbinProbes.cfg

GrubbyZebra committed:
Update KerbinProbes.cfg

GrubbyZebra committed:
Add files via upload

GrubbyZebra committed:
Add files via upload

jrossignol commented:
Looks good at a glance.  Will have to spend some time taking a closer look at it (hopefully over the next few days).

GrubbyZebra committed:
Update KerbinProgram.cfg

Added the Gemini Missions

GrubbyZebra committed:
Update CrewedMissions-1-KerbinProgram.cfg

Edited RSS Title and Contract Message.

jrossignol commented:
I think everything looks good from a technical standpoint.  My concern is that the contract asks too much.  Specifically:

1. The first part (break the sound barrier) feels like it's an x-plane type of requirement (which doesn't necessarily fit).  But it doesn't really have any "plane" restrictions (parts, horizontal velocity check or altitude ceiling), so the player could end up doing this one and reach the upper atmosphere with the same vessel (one after the other because of the New Vessel requirement).   So I say drop this (but consider folding into an entirely separate x-plane strategy...

1. The last two Rendezvous + Dock feel like too forcing the player too much down a path they may not want to go down.  If I wanted to rush to the Mun, but wanted to take this as the "early" strategy then I don't think I'd want to do these (and I definitely wouldn't want to have to Rendezvous and Dock seperately).  My suggestion here is a whole new strategy (ie. Gemini analogue).

1. The NewVessel parameter may not be doing what you think - it means "the vessel must be launched after the contract is accepted".  So technically, one vessel could do all the different parameters in the contract (although the RecoverVessel actually ends up making this impossible).  But I don't think this is necessarily bad - I would remove the NewVessel/RecoverVessel and embrace that - it gives the player more flexibility which feels more like what Strategia should be to me.  To avoid cheesing though, I'd change the Strategy requirement from "not having ReturnFromOrbitManned" to "not having OrbitManned".

1. If we embrace the "one vessel can do all the steps", then I'd suggest splitting the EVA + 1 week into different steps.

Let me know what you think.

GrubbyZebra committed:
Update KerbinProgram.cfg

Change SoundBarrier to FirstLaunch

fwdixon commented:
No worries. 

1) I wanted to keep the "first launch" and "high altitude" as separate goals for folks playing stock, but agree that the "sound barrier" verbiage makes it sound like a plane contract and not a rocket.  SoundBarrier has been replaced by FirstLaunch in the contract.

2) Ideally I wanted the Gemini contract to be a separate strategy from the Mercury stuff, and have it be optional to the moon programs. I couldn't get it to work properly under the Kerbin Program, but thinking about it this morning I think I can get it as a +1 under the Moon Program strategies. This give the player the most flexibility as they can either choose to start exploring the moons immediately after orbit or build (and get some bonus awards) up to an Apollo-style mission.

3) I realized on my play-thru that the new vessel only needs to be in once per contract, but forgot to go back and strip the duplicates out. Will take care of that tonight (trying to use the github editor at work is a pain).

4) I'm torn on this one. The one-week is really only needed if the player has a life support mod installed. And having the EVA on the 1 week requirement is there to give the player

jrossignol commented:
Sounds good - on the LS one we could just do a NEEDS[TacLifeSupport|USILifeSupport] for that node.  Not sure if there are others we'd want to support?  I assume Kerbalism is so out there it wouldn't play with Strategia well anyway...

GrubbyZebra committed:
Update KerbinProgram.cfg

Removed redundant NewVessel Parameters (left the one in for the 2-crew launch due to future development plans). 

Added requirements for TACLS, USILS, and Snacks for the 7d endurance parameter.

fwdixon commented:
TACLS, USILS, and Snacks added to the 7d endurance parameter. Left the 2-crew launch and EVA in place.

fwdixon commented:
finally got my new computer set up and can start back on this. 

One thing I noticed inadvertently today is that Kerbin Probes doesn't check for the "first launch" milestone but has a reward for "first launch (just a contract param, not the milestone". Wondering if I should require no launches to take the strategy or reword the contract parameter....

jrossignol commented:
I'd go with rewording the contract parameter - limiting it to before first launch is extremely strict.

fwdixon commented:
Been playing through Kerbin probes with different parts packs, and have realized that the requires science and requires 2 electricity is too restrictive (e.g. BDB probes have science experiments in them that aren't registered by the parameter). Will strip those out of the contract but leave in the transmit or recover parameter.

fwdixon commented:
I don't think there is a limit to before first launch. It should be limited to before first orbit (since the ultimate goal of the strategy/contract is first orbit). 

Or are you saying that calling the parameter "First Launch" is confusing?

GrubbyZebra committed:
Merge remote-tracking branch 'upstream/master'
